cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -68,23 +68,34 @@
 
   /**
    * Adds a new metadata entry for a particular partition.
-   * If the metadata key already exists, it will be overwritten.
+   * Note that existing entries cannot be updated.
    *
+   * @throws DataSetException when an attempt is made to update an existing entry
    * @throws PartitionNotFoundException when a partition for the given key is not found
    * @throws IllegalArgumentException if the partition key does not match the partitioning of the dataset
    */
   void addMetadata(PartitionKey key, String metadataKey, String metadataValue);
 
   /**
    * Adds a set of new metadata entries for a particular partition.
-   * If the metadata key already exists, it will be overwritten.
+   * Note that existing entries cannot be updated.
    *
+   * @throws DataSetException when an attempt is made to update existing entries
    * @throws PartitionNotFoundException when a partition for the given key is not found
    * @throws IllegalArgumentException if the partition key does not match the partitioning of the dataset
    */
   void addMetadata(PartitionKey key, Map<String, String> metadata);
 
   /**
+   * Sets metadata entries for a particular partition. If the metadata entry key does not already exist, it will be
+   * created; otherwise, it will be overwritten.
```

Comment:
This updates each key by itself, right? Upsert-like. Is it worth mentioning that other existing keys remain unchanged? Also, is there an API to remove a metadata key?